lib/ansible/modules/windows/win_psmodule.ps1

```diff
@@ -1,186 +1,380 @@
 $repo = Get-AnsibleParam -obj $params -name "repository" -type "str"
-$url = Get-AnsibleParam -obj $params -name "url" -type "str"
-$state = Get-AnsibleParam -obj $params -name "state" -type "str" -default "present" -validateset "present", "absent"
+$state = Get-AnsibleParam -obj $params -name "state" -type "str" -default "present" -validateset "present", "absent", "latest"
 $allow_clobber = Get-AnsibleParam -obj $params -name "allow_clobber" -type "bool" -default $false
-$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -default $false
+$skip_publisher_check = Get-AnsibleParam -obj $params -name "skip_publisher_check" -type "bool" -default $false
+$allow_prerelease = Get-AnsibleParam -obj $params -name "allow_prerelease" -type "bool" -default $false
 
-$result = @{"changed" = $false
-            "output" = ""
-            "nuget_changed" = $false
-            "repository_changed" = $false}
+$result = @{changed = $false}
 
 Function Install-NugetProvider {
-  param(
-    [bool]$CheckMode
+    Param(
+        [Bool]$CheckMode
     )
-  $PackageProvider = Get-PackageProvider -ListAvailable|?{($_.name -eq 'Nuget') -and ($_.version -ge "2.8.5.201")}
-  if (!($PackageProvider)){
-      try{
-        Install-PackageProvider -Name NuGet -MinimumVersion 2.8.5.201 -Force -ErrorAction Stop -WhatIf:$CheckMode | out-null
-        $result.changed = $true
-        $result.nuget_changed = $true
-      }
-      catch{
-        $ErrorMessage = "Problems adding package provider: $($_.Exception.Message)"
-        Fail-Json $result $ErrorMessage
-      }
+    $PackageProvider = Get-PackageProvider -ListAvailable | Where-Object {($_.name -eq 'Nuget') -and ($_.version -ge "2.8.5.201")}
+    if (-not($PackageProvider)){
+        try{
+            Install-PackageProvider -Name NuGet -MinimumVersion 2.8.5.201 -Force -WhatIf:$CheckMode | out-null
+            $result.changed = $true
+        }
+        catch{
+            $ErrorMessage = "Problems adding package provider: $($_.Exception.Message)"
+            Fail-Json $result $ErrorMessage
+        }
     }
 }
 
-Function Install-Repository {
+Function Install-PrereqModule {
     Param(
-    [Parameter(Mandatory=$true)]
-    [string]$Name,
-    [Parameter(Mandatory=$true)]
-    [string]$Url,
-    [bool]$CheckMode
+        [Bool]$CheckMode
     )
-    $Repo = (Get-PSRepository).SourceLocation
-
-    # If repository isn't already present, try to register it as trusted.
-    if ($Repo -notcontains $Url){ 
-      try {
-           if (!($CheckMode)) {
-               Register-PSRepository -Name $Name -SourceLocation $Url -InstallationPolicy Trusted -ErrorAction Stop       
-           }
-          $result.changed = $true
-          $result.repository_changed = $true
-      }
-      catch {
-        $ErrorMessage = "Problems adding $($Name) repository: $($_.Exception.Message)"
-        Fail-Json $result $ErrorMessage
-      }
+
+    # Those are minimum required versions of modules.
```

Comment:
Why are these the minimum? I've been able to use this module using the versions included with PSv5 which are at `1.0.0.1`?

Reply:
The PowerShellGet module v. 1.6.0 introduced support for prereleases.
The PackageManagement module v. 1.1.7.0 [is required by the PowerShellGet](https://github.com/PowerShell/PowerShellGet/blob/4c96ddba0c3c98318b4f546ef9f78cfb1400c514/PowerShellGet/PowerShellGet.psd1#L44) so I added that explicitly.